.github/workflows/summarize-checks.yaml

```diff
@@ -0,0 +1,49 @@
+name: "[TEST-IGNORE] Summarize Checks"
+
+on:
+  # We care about more labels than those that trigger a workflow run right? As in there are labels added that should update
+  # next steps to merge, but those labels might not trigger one of our status update runs.
+  # So we run this workflow on any label added to a PR.
+  workflow_run:
+    # todo, I feel like I'm missing a couple. I recall 6 being necessary/present.
+    workflows:
+    - "\\[TEST-IGNORE\\] Swagger SemanticValidation - Set Status"
+    - "\\[TEST-IGNORE\\] Swagger ModelValidation - Set Status"
+    - "\\[TEST-IGNORE\\] Swagger Avocado - Set Status"
+    - "Swagger LintDiff - Set Status"
+    - "SDK Validation Status"
+    types: [completed]
+  pull_request_target: # when a PR is labeled. NOT pull_request, because that would run on the PR branch, not the base branch.
+    types:
+      - labeled
+
+permissions:
+  actions:       read   # to inspect workflow_run metadata
+  contents:      read   # for actions/checkout
+  checks:        read   # for octokit.rest.checks.listForRef
+  statuses:      read   # for octokit.rest.repos.getCombinedStatusForRef
+  issues:        write  # to post comments via the Issues API I'm not certain if I need this one or pull-requests only. we'll find out!
+  pull-requests: write  # to post comments via the Pull Requests API
+
+jobs:
+  run-summarize-checks:
+    name: "[TEST-IGNORE] Summarize Checks"
+    runs-on: ubuntu-24.04
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+
+      - name: Setup Node and install deps
```

Comment:
For performance, I think we should try to implement this workflow in pure JS under .github/workflows/src.  Example:

https://github.com/Azure/azure-rest-api-specs/blob/main/.github/workflows/arm-auto-signoff.yaml
https://github.com/Azure/azure-rest-api-specs/blob/main/.github/workflows/src/arm-auto-signoff.js

Reply:
Ok. I'll refactor.